components/camera/videosource/helpers.go

```diff
@@ -83,18 +98,17 @@ func readDepthURL(ctx context.Context, client http.Client, url string, immediate
 	if err != nil {
 		return nil, errors.Wrap(err, "couldn't ready depth url")
 	}
-
-	if !immediate {
-		if lazyImg, ok, err := checkLazyFromData(ctx, depthData); err != nil {
-			return nil, err
-		} else if ok {
-			return lazyImg, nil
-		}
+	mimeType, err := checkMimeType(ctx, depthData)
+	if err != nil {
+		return nil, err
 	}
-
-	img, err := decodeImage(depthData)
+	actualMimeType, isLazy := utils.CheckLazyMIMEType(mimeType)
+	if !immediate && isLazy {
+		return rimage.NewLazyEncodedImage(depthData, actualMimeType), nil
```

Comment:
[opt] Since DecodeImage will create the lazy encoded image, you could move the check for `!immediate && isLazy` below the call to DecodeImage, and just return img. This would make the function more similar to readColorURL().